src/lib/getIgnoredUpgrades.ts

```diff
@@ -13,36 +14,54 @@ export async function getIgnoredUpgrades(
   upgradedPeerDependencies: Index<Index<Version>>,
   options: Options = {},
 ) {
+  const upgradedPackagesWithPeerRestriction = Object.fromEntries(
+    Object.entries({
+      ...current,
+      ...upgraded,
+    }).map(([packageName, versionSpec]) => {
+      return [packageName, semver.minVersion(versionSpec)?.version ?? versionSpec]
+    }),
+  )
   const [upgradedLatestVersions, latestVersionResults] = await upgradePackageDefinitions(current, {
     ...options,
     peer: false,
     peerDependencies: undefined,
     loglevel: 'silent',
   })
-
+  const upgradedPeerDependenciesLatest = await getPeerDependenciesFromRegistry(upgradedLatestVersions, options)
   return Object.entries(upgradedLatestVersions)
     .filter(([pkgName, newVersion]) => upgraded[pkgName] !== newVersion)
-    .reduce(
-      (accum, [pkgName, newVersion]) => ({
+    .reduce((accum, [pkgName, newVersion]) => {
+      let reason = Object.entries(upgradedPeerDependencies)
+        .filter(
+          ([, peers]) =>
+            peers[pkgName] !== undefined &&
+            latestVersionResults[pkgName]?.version &&
+            !satisfies(latestVersionResults[pkgName].version!, peers[pkgName]),
+        )
+        .reduce((accumReason, [peerPkg, peers]) => ({ ...accumReason, [peerPkg]: peers[pkgName] }), {} as Index<string>)
+      if (Object.keys(reason).length === 0) {
+        const peersOfPkg = upgradedPeerDependenciesLatest?.[pkgName] || {}
+        reason = Object.entries(peersOfPkg)
+          .filter(
+            ([peer, peerSpec]) =>
+              upgradedPackagesWithPeerRestriction[peer] &&
+              !satisfies(upgradedPackagesWithPeerRestriction[peer], peerSpec),
+          )
```

Comment:
I'm using the same logic as in:
https://github.com/raineorshine/npm-check-updates/blob/0f17df28573aea128f40335e9856526f619db9e3/src/lib/runLocal.ts#L183-L192

I don't think intersect is the right way to go as it is not sufficient to make sure that peer dependency is met. For example, if package P1 spec is `^1.0.0` and P2 has P1 as peer with spec `^1.2.0`, the 2 ranges will intersect  but it will not mean that the peer dependency is met...

`current` is not sufficient, as we need to check against the versions that are currently upgraded.

Thoughts?

Reply:
But if the P1 spec is `^1.0.0` (not `1.0.0`) then npm will install `1.2.0` or later and the peer dependency will be met, no?

Reply:
That will not be the case if your lcokfile was not updated and P1 was not upgraded due to a filter. But if you think that will be better I will make the change, looking forward to your opinion

Reply:
npm-check-updates is generally just concerned with package.json versions, and assumes that the user will run `npm install` and manage the lockfile through npm. The lockfile is basically always out of date after running `ncu -u`, so I don't see that as a problem. It would be a problem if npm-check-updates upgraded a version range to something that was incompatible though. If you think that's the case, please let me know. I didn't notice that the existing code uses the same `minVersion` trick to get the bottom of the range.

I'm having a hard time picturing the situation you described with a filter on P1. If you think that's a problem, we should add a unit test that covers that. It is admittedly difficult to reason about this abstractly, at least for me.

Reply:
Im not sure I can write a test for it. I changed to intresect in both places where I used minVersion